// @hakit/core/dist/es/hooks/useService/index.js
import { useMemo as l, useCallback as m } from "react";
import { uniq as v } from "lodash";
import { W as S, X as $ } from "../../../index-1tDtVGr2.js";
import "home-assistant-js-websocket";
import "@iconify/react";
import "use-debounce";
import "../useLocale/locales/index.js";
import "../../utils/date.js";
function a(e, r, t) {
  return new Proxy(
    // @ts-expect-error - purposely not defining the target object here
    {},
    {
      get: (s, n) => {
        const i = n;
        if (i !== "toJSON")
          return function(p) {
            const { target: f, serviceData: o, returnResponse: u } = p || {};
            let c = t ?? f;
            return Array.isArray(c) && (c = [...v(c)]), console.info(
              `${$("perform_action", {
                search: "{name}",
                replace: `${e}.${i}`
              })} ${e}.${i}:`,
              {
                target: c,
                serviceData: o,
                returnResponse: u
              }
            ), r({
              domain: e,
              service: i,
              serviceData: o,
              target: c,
              // necessary cast here as the overloads expect true | false | undefined not a boolean
              returnResponse: u
            });
          };
      }
    }
  );
}
function b(e, r) {
  const { callService: t } = S.getState().helpers, s = l(() => e ? a(e, t, r) : void 0, [e, t, r]), n = m(
    (i) => a(i, t, r),
    [t, r]
  );
  return e ? s : n;
}
export {
  a as createService,
  b as useService
};
