// @hakit/core/dist/es/hooks/useHistory/index.js
import { useRef as C, useState as m, useMemo as f, useEffect as h } from "react";
import { useSubscribeEntity as H } from "../useSubscribeEntity/index.js";
import { W as S } from "../../../index-1tDtVGr2.js";
import { subscribeHistory as p, computeHistory as O } from "./history.js";
import { coordinatesMinimalResponseCompressedState as R } from "./coordinates.js";
const E = (i, e) => {
  const a = S((t) => t.connection), n = C(!1), l = H(i), [o, y] = m({}), [u, g] = m({
    loading: !e?.disable,
    timeline: [],
    entityHistory: [],
    coordinates: []
  }), s = f(() => ({
    disable: e?.disable,
    significantChangesOnly: e?.significantChangesOnly,
    minimalResponse: e?.minimalResponse,
    hoursToShow: e?.hoursToShow,
    limits: e?.limits,
    forceNumeric: e?.forceNumeric,
    splitDeviceClasses: e?.splitDeviceClasses
  }), [
    e?.disable,
    e?.significantChangesOnly,
    e?.minimalResponse,
    e?.hoursToShow,
    e?.limits,
    e?.forceNumeric,
    e?.splitDeviceClasses
  ]);
  return h(() => {
    if (!a || s?.disable) return;
    let t = !0, c = null;
    return p({
      connection: a,
      entityIds: [i],
      significantChangesOnly: s?.significantChangesOnly,
      minimalResponse: s?.minimalResponse,
      hoursToShow: s?.hoursToShow,
      callbackFunction: (r) => {
        t && (n.current = !0, y(r));
      }
    }).then((r) => {
      c = r;
    }).catch(() => {
      c?.(), n.current = !1;
    }), () => {
      t = !1, c?.(), n.current = !1;
    };
  }, [i, s, a]), h(() => {
    if (!s?.disable && n.current) {
      const t = l(!0);
      if (!t) return;
      const r = O(t, o, s.forceNumeric, s?.splitDeviceClasses).timeline.filter(({ entity_id: b }) => b === i), d = R(
        o[i],
        s?.hoursToShow ?? 24,
        500,
        // viewbox of the svgGraph
        typeof s?.significantChangesOnly > "u" || s?.significantChangesOnly === !0 ? 1 : 2,
        s?.limits
      ) ?? [];
      g({
        loading: !1,
        timeline: r.length > 0 ? r[0].data : [],
        entityHistory: o[i],
        coordinates: d
      });
    }
  }, [i, s, l, o]), f(() => u, [u]);
};
export {
  E as useHistory
};
