// @hakit/core/dist/es/utils/colors.js
import { rgb2hex as i, hs2rgb as g } from "./colors/convert-color.js";
import { temperature2rgb as l, rgbw2rgb as _, rgbww2rgb as n } from "./colors/convert-light-color.js";
import { stateColorBrightness as m } from "./colors/index.js";
import { c as h } from "../../index-1tDtVGr2.js";
function p(r) {
  if (r.attributes) {
    if ("hs_color" in r.attributes && r.attributes.hs_color !== null)
      return g([r.attributes.hs_color[0], r.attributes.hs_color[1] / 100]);
    if ("color_temp_kelvin" in r.attributes && r.attributes.color_temp_kelvin !== null)
      return l(r.attributes.color_temp_kelvin);
    if ("rgb_color" in r.attributes && r.attributes.rgb_color !== null)
      return r.attributes.rgb_color;
    if ("rgbw_color" in r.attributes && r.attributes.rgbw_color !== null)
      return _(r.attributes.rgbw_color);
    if ("rgbww_color" in r.attributes && r.attributes.rgbww_color !== null)
      return n(r.attributes.rgbww_color, r.attributes.min_color_temp_kelvin, r.attributes.max_color_temp_kelvin);
    if (h(r))
      return l(2e3);
  }
  return null;
}
function k(r) {
  const o = r ? p(r) : null, t = o ? o.map((c) => c.toFixed(0)).map(Number) : null, s = o ? i(o) : "var(--ha-A400)", u = t ? `rgba(${t.join(", ")})` : "var(--ha-S500-contrast)", b = t ? `rgba(${[...t, 0.35].join(", ")})` : "var(--ha-A200)", { css: a, raw: e } = m(r);
  return {
    color: o || [33, 33, 33],
    hexColor: s,
    rgbColor: u,
    rgbaColor: b,
    brightness: a,
    brightnessValue: e
  };
}
export {
  k as getCssColorValue
};
