// @hakit/core/dist/es/utils/index.js
import { rgbw2rgb as r, rgbww2rgb as a, temperature2rgb as o } from "./colors/convert-light-color.js";
import { hex2rgb as m, hs2rgb as s, hsv2rgb as n, lab2hex as p, lab2rgb as y, rgb2hex as u, rgb2hs as g, rgb2hsv as f, rgb2lab as E } from "./colors/convert-color.js";
import { COLORS as D, getColorByIndex as c } from "./colors/colors.js";
import { THEME_COLORS as h, computeCssColor as b } from "./colors/compute-color.js";
import { expandHex as N, hexBlend as d } from "./colors/hex.js";
import { getRGBContrastRatio as C, luminosity as R, rgbContrast as A } from "./colors/rgb.js";
import { stateColorBrightness as B } from "./colors/index.js";
import { labBrighten as F, labDarken as M } from "./colors/lab.js";
import { D as v, N as L, O as U, T as W, r as k, Q as P, e as w, n as V, m as Y, p as H, o as Z, v as j, t as G, y as q, q as z, f as J, R as K, S as Q, i as X, h as $, I as tt, J as et, u as rt, E as at, x as ot, w as it, L as mt, M as st, j as nt, k as pt, K as yt, A as ut, z as gt, g as ft, P as Et, a as lt, c as Dt, b as ct, l as Tt, d as ht, C as bt, G as xt, s as Nt, F as dt, B as St } from "../../index-1tDtVGr2.js";
import { getCssColorValue as Rt } from "./colors.js";
import { computeDomain as It } from "./computeDomain.js";
import { computeDomainTitle as Ot } from "./computeDomainTitle.js";
import { supportsFeatureFromAttributes as Mt } from "./supports-feature.js";
import { timeAgo as vt } from "./time/time-ago.js";
import { caseInsensitiveStringCompare as Ut, stringCompare as Wt, toReadableString as kt } from "./string.js";
import { formatNumber as wt, getDefaultFormatOptions as Vt, getNumberFormatOptions as Yt, isNumericFromAttributes as Ht, isNumericState as Zt, round as jt } from "./number.js";
import { LOCAL_TIME_ZONE as qt, TimeFormat as zt, TimeZone as Jt, UNIT_TO_MILLISECOND_CONVERT as Kt, checkValidDate as Qt, formatAmPmSuffix as Xt, formatDate as $t, formatDateMonth as te, formatDateMonthYear as ee, formatDateNumeric as re, formatDateShort as ae, formatDateTime as oe, formatDateTimeNumeric as ie, formatDateTimeWithBrowserDefaults as me, formatDateTimeWithSeconds as se, formatDateVeryShort as ne, formatDateWeekday as pe, formatDateWeekdayDay as ye, formatDateWeekdayShort as ue, formatDateYear as ge, formatDuration as fe, formatHour as Ee, formatMinute as le, formatSeconds as De, formatShortDateTime as ce, formatShortDateTimeWithConditionalYear as Te, formatShortDateTimeWithYear as he, formatTime as be, formatTimeWithoutAmPm as xe, isDate as Ne, isTimestamp as de, millisecondsToDuration as Se, resolveTimeZone as Ce, shouldUseAmPm as Re } from "./date.js";
import { stripPrefixFromEntityName as Ie } from "./stripPrefixFromEntityName.js";
export {
  D as COLORS,
  v as DEFAULT_ENTITY_NAME,
  L as DOMAIN_ATTRIBUTES_FORMATTERS,
  U as DOMAIN_ATTRIBUTES_UNITS,
  qt as LOCAL_TIME_ZONE,
  W as TEMPERATURE_ATTRIBUTES,
  h as THEME_COLORS,
  zt as TimeFormat,
  Jt as TimeZone,
  Kt as UNIT_TO_MILLISECOND_CONVERT,
  Ut as caseInsensitiveStringCompare,
  Qt as checkValidDate,
  k as computeAreaName,
  P as computeAttributeValueDisplay,
  b as computeCssColor,
  w as computeDefaultFavoriteColors,
  V as computeDeviceName,
  Y as computeDeviceNameDisplay,
  It as computeDomain,
  Ot as computeDomainTitle,
  H as computeEntityEntryName,
  Z as computeEntityName,
  j as computeEntityNameDisplay,
  G as computeEntityNameList,
  q as computeEntityRegistryName,
  z as computeFloorName,
  J as computeObjectId,
  K as computeStateDisplay,
  Q as computeStateDisplayFromEntityAttributes,
  X as computeStateName,
  $ as computeStateNameFromEntityAttributes,
  tt as entityRegistryByEntityId,
  et as entityRegistryById,
  rt as entityUseDeviceName,
  N as expandHex,
  at as fetchEntityRegistryDisplay,
  ot as findBatteryChargingEntity,
  it as findBatteryEntity,
  Xt as formatAmPmSuffix,
  $t as formatDate,
  te as formatDateMonth,
  ee as formatDateMonthYear,
  re as formatDateNumeric,
  ae as formatDateShort,
  oe as formatDateTime,
  ie as formatDateTimeNumeric,
  me as formatDateTimeWithBrowserDefaults,
  se as formatDateTimeWithSeconds,
  ne as formatDateVeryShort,
  pe as formatDateWeekday,
  ye as formatDateWeekdayDay,
  ue as formatDateWeekdayShort,
  ge as formatDateYear,
  fe as formatDuration,
  Ee as formatHour,
  le as formatMinute,
  wt as formatNumber,
  De as formatSeconds,
  ce as formatShortDateTime,
  Te as formatShortDateTimeWithConditionalYear,
  he as formatShortDateTimeWithYear,
  be as formatTime,
  xe as formatTimeWithoutAmPm,
  mt as getAutomaticEntityIds,
  c as getColorByIndex,
  Rt as getCssColorValue,
  Vt as getDefaultFormatOptions,
  st as getEntities,
  nt as getEntityContext,
  pt as getEntityEntryContext,
  yt as getEntityPlatformLookup,
  ut as getExtendedEntityRegistryEntries,
  gt as getExtendedEntityRegistryEntry,
  ft as getLightCurrentModeRgbColor,
  Yt as getNumberFormatOptions,
  C as getRGBContrastRatio,
  Et as getWeatherUnit,
  m as hex2rgb,
  d as hexBlend,
  s as hs2rgb,
  n as hsv2rgb,
  Ne as isDate,
  Ht as isNumericFromAttributes,
  Zt as isNumericState,
  de as isTimestamp,
  p as lab2hex,
  y as lab2rgb,
  F as labBrighten,
  M as labDarken,
  lt as lightIsInColorMode,
  Dt as lightSupportsBrightness,
  ct as lightSupportsColor,
  Tt as lightSupportsColorMode,
  ht as lightSupportsFavoriteColors,
  R as luminosity,
  Se as millisecondsToDuration,
  bt as removeEntityRegistryEntry,
  Ce as resolveTimeZone,
  u as rgb2hex,
  g as rgb2hs,
  f as rgb2hsv,
  E as rgb2lab,
  A as rgbContrast,
  r as rgbw2rgb,
  a as rgbww2rgb,
  jt as round,
  Re as shouldUseAmPm,
  xt as sortEntityRegistryByName,
  Nt as stateActive,
  B as stateColorBrightness,
  Wt as stringCompare,
  Ie as stripPrefixFromEntityName,
  dt as subscribeEntityRegistryDisplay,
  Mt as supportsFeatureFromAttributes,
  o as temperature2rgb,
  vt as timeAgo,
  kt as toReadableString,
  St as updateEntityRegistryEntry
};
