// @hakit/core/dist/es/utils/number.js
const u = (i, t) => {
  const r = t?.display_precision;
  if (r != null)
    return {
      maximumFractionDigits: r,
      minimumFractionDigits: r
    };
  if (Number.isInteger(Number(i?.attributes?.step)) && Number.isInteger(Number(i?.state)))
    return { maximumFractionDigits: 0 };
}, a = (i) => m(i.attributes), m = (i, t) => !!i.unit_of_measurement || !!i.state_class || (t || []).includes(i.device_class || ""), s = (i, t) => {
  const r = {
    maximumFractionDigits: 2,
    ...t
  };
  if (typeof i != "string")
    return r;
  if (!t || t.minimumFractionDigits === void 0 && t.maximumFractionDigits === void 0) {
    const e = i.indexOf(".") > -1 ? i.split(".")[1].length : 0;
    r.minimumFractionDigits = e, r.maximumFractionDigits = e;
  }
  return r;
}, n = (i, t = 2) => Math.round(i * 10 ** t) / 10 ** t, c = (i, t) => (Number.isNaN = Number.isNaN || function r(e) {
  return typeof e == "number" && r(e);
}, !Number.isNaN(Number(i)) && i !== "" ? new Intl.NumberFormat(
  "en-US",
  s(i, {
    ...t,
    useGrouping: !1
  })
).format(Number(i)) : typeof i == "string" ? i : `${n(i, t?.maximumFractionDigits).toString()}${t?.style === "currency" ? ` ${t.currency}` : ""}`);
export {
  c as formatNumber,
  s as getDefaultFormatOptions,
  u as getNumberFormatOptions,
  m as isNumericFromAttributes,
  a as isNumericState,
  n as round
};
