// @hakit/components/dist/es/Cards/SensorCard/index.js
import { j as t } from "../../../emotion-react-jsx-runtime.browser.esm-BErL7L7Y.js";
import { c } from "../../../emotion-styled-base.browser.esm-Dyg8Y4aq.js";
import { useHass as p } from "@hakit/core";
import { ErrorBoundary as i } from "react-error-boundary";
import { f as u, B as S } from "../../../ThemeControlsModal-gl0xLRqL.js";
const f = /* @__PURE__ */ c(S, process.env.NODE_ENV === "production" ? {
  target: "e16wpwvj0"
} : {
  target: "e16wpwvj0",
  label: "StyledSensorCard"
})();
function y({
  entity: r,
  className: e,
  cssStyles: o,
  hideToggle: s = !0,
  service: a,
  serviceData: n,
  ...l
}) {
  const d = p((m) => m.globalComponentStyles);
  return /* @__PURE__ */ t(
    f,
    {
      disableActiveState: !0,
      hideToggle: s,
      service: a,
      serviceData: n,
      graph: {
        entity: r,
        // content spacing is 1rem, so we offset it here by the same amount
        adjustGraphSpaceBy: "1rem"
      },
      entity: r,
      className: `sensor-card ${e ?? ""}`,
      cssStyles: `
        ${d?.sensorCard ?? ""}
        ${o ?? ""}
      `,
      ...l
    }
  );
}
function v(r) {
  const e = {
    xxs: 12,
    xs: 6,
    sm: 6,
    md: 4,
    lg: 4,
    xlg: 3
  };
  return /* @__PURE__ */ t(i, { ...u({
    prefix: "SensorCard"
  }), children: /* @__PURE__ */ t(y, { ...e, ...r }) });
}
export {
  v as SensorCard
};
