// @hakit/components/dist/emotion-react-jsx-runtime.browser.esm-BErL7L7Y.js
import * as A from "react/jsx-runtime";
import * as b from "react";
import { forwardRef as de, useContext as ye } from "react";
import ve from "@emotion/cache";
import { getRegisteredStyles as Se, registerStyles as Ee, insertStyles as pe } from "@emotion/utils";
import { serializeStyles as Te } from "@emotion/serialize";
import { u as _e } from "./emotion-use-insertion-effect-with-fallbacks.browser.esm-o5bKZLsy.js";
var M = { exports: {} }, n = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var L;
function Pe() {
  if (L) return n;
  L = 1;
  var r = typeof Symbol == "function" && Symbol.for, s = r ? Symbol.for("react.element") : 60103, i = r ? Symbol.for("react.portal") : 60106, o = r ? Symbol.for("react.fragment") : 60107, c = r ? Symbol.for("react.strict_mode") : 60108, y = r ? Symbol.for("react.profiler") : 60114, m = r ? Symbol.for("react.provider") : 60109, v = r ? Symbol.for("react.context") : 60110, S = r ? Symbol.for("react.async_mode") : 60111, u = r ? Symbol.for("react.concurrent_mode") : 60111, T = r ? Symbol.for("react.forward_ref") : 60112, _ = r ? Symbol.for("react.suspense") : 60113, $ = r ? Symbol.for("react.suspense_list") : 60120, E = r ? Symbol.for("react.memo") : 60115, d = r ? Symbol.for("react.lazy") : 60116, P = r ? Symbol.for("react.block") : 60121, x = r ? Symbol.for("react.fundamental") : 60117, C = r ? Symbol.for("react.responder") : 60118, R = r ? Symbol.for("react.scope") : 60119;
  function l(e) {
    if (typeof e == "object" && e !== null) {
      var p = e.$$typeof;
      switch (p) {
        case s:
          switch (e = e.type, e) {
            case S:
            case u:
            case o:
            case y:
            case c:
            case _:
              return e;
            default:
              switch (e = e && e.$$typeof, e) {
                case v:
                case T:
                case d:
                case E:
                case m:
                  return e;
                default:
                  return p;
              }
          }
        case i:
          return p;
      }
    }
  }
  function f(e) {
    return l(e) === u;
  }
  return n.AsyncMode = S, n.ConcurrentMode = u, n.ContextConsumer = v, n.ContextProvider = m, n.Element = s, n.ForwardRef = T, n.Fragment = o, n.Lazy = d, n.Memo = E, n.Portal = i, n.Profiler = y, n.StrictMode = c, n.Suspense = _, n.isAsyncMode = function(e) {
    return f(e) || l(e) === S;
  }, n.isConcurrentMode = f, n.isContextConsumer = function(e) {
    return l(e) === v;
  }, n.isContextProvider = function(e) {
    return l(e) === m;
  }, n.isElement = function(e) {
    return typeof e == "object" && e !== null && e.$$typeof === s;
  }, n.isForwardRef = function(e) {
    return l(e) === T;
  }, n.isFragment = function(e) {
    return l(e) === o;
  }, n.isLazy = function(e) {
    return l(e) === d;
  }, n.isMemo = function(e) {
    return l(e) === E;
  }, n.isPortal = function(e) {
    return l(e) === i;
  }, n.isProfiler = function(e) {
    return l(e) === y;
  }, n.isStrictMode = function(e) {
    return l(e) === c;
  }, n.isSuspense = function(e) {
    return l(e) === _;
  }, n.isValidElementType = function(e) {
    return typeof e == "string" || typeof e == "function" || e === o || e === u || e === y || e === c || e === _ || e === $ || typeof e == "object" && e !== null && (e.$$typeof === d || e.$$typeof === E || e.$$typeof === m || e.$$typeof === v || e.$$typeof === T || e.$$typeof === x || e.$$typeof === C || e.$$typeof === R || e.$$typeof === P);
  }, n.typeOf = l, n;
}
var a = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var z;
function Re() {
  return z || (z = 1, process.env.NODE_ENV !== "production" && (function() {
    var r = typeof Symbol == "function" && Symbol.for, s = r ? Symbol.for("react.element") : 60103, i = r ? Symbol.for("react.portal") : 60106, o = r ? Symbol.for("react.fragment") : 60107, c = r ? Symbol.for("react.strict_mode") : 60108, y = r ? Symbol.for("react.profiler") : 60114, m = r ? Symbol.for("react.provider") : 60109, v = r ? Symbol.for("react.context") : 60110, S = r ? Symbol.for("react.async_mode") : 60111, u = r ? Symbol.for("react.concurrent_mode") : 60111, T = r ? Symbol.for("react.forward_ref") : 60112, _ = r ? Symbol.for("react.suspense") : 60113, $ = r ? Symbol.for("react.suspense_list") : 60120, E = r ? Symbol.for("react.memo") : 60115, d = r ? Symbol.for("react.lazy") : 60116, P = r ? Symbol.for("react.block") : 60121, x = r ? Symbol.for("react.fundamental") : 60117, C = r ? Symbol.for("react.responder") : 60118, R = r ? Symbol.for("react.scope") : 60119;
    function l(t) {
      return typeof t == "string" || typeof t == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      t === o || t === u || t === y || t === c || t === _ || t === $ || typeof t == "object" && t !== null && (t.$$typeof === d || t.$$typeof === E || t.$$typeof === m || t.$$typeof === v || t.$$typeof === T || t.$$typeof === x || t.$$typeof === C || t.$$typeof === R || t.$$typeof === P);
    }
    function f(t) {
      if (typeof t == "object" && t !== null) {
        var w = t.$$typeof;
        switch (w) {
          case s:
            var g = t.type;
            switch (g) {
              case S:
              case u:
              case o:
              case y:
              case c:
              case _:
                return g;
              default:
                var D = g && g.$$typeof;
                switch (D) {
                  case v:
                  case T:
                  case d:
                  case E:
                  case m:
                    return D;
                  default:
                    return w;
                }
            }
          case i:
            return w;
        }
      }
    }
    var e = S, p = u, N = v, F = m, K = s, X = T, B = o, G = d, J = E, Z = i, Q = y, k = c, ee = _, Y = !1;
    function re(t) {
      return Y || (Y = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), j(t) || f(t) === S;
    }
    function j(t) {
      return f(t) === u;
    }
    function te(t) {
      return f(t) === v;
    }
    function oe(t) {
      return f(t) === m;
    }
    function ne(t) {
      return typeof t == "object" && t !== null && t.$$typeof === s;
    }
    function ae(t) {
      return f(t) === T;
    }
    function se(t) {
      return f(t) === o;
    }
    function ie(t) {
      return f(t) === d;
    }
    function ce(t) {
      return f(t) === E;
    }
    function fe(t) {
      return f(t) === i;
    }
    function ue(t) {
      return f(t) === y;
    }
    function le(t) {
      return f(t) === c;
    }
    function me(t) {
      return f(t) === _;
    }
    a.AsyncMode = e, a.ConcurrentMode = p, a.ContextConsumer = N, a.ContextProvider = F, a.Element = K, a.ForwardRef = X, a.Fragment = B, a.Lazy = G, a.Memo = J, a.Portal = Z, a.Profiler = Q, a.StrictMode = k, a.Suspense = ee, a.isAsyncMode = re, a.isConcurrentMode = j, a.isContextConsumer = te, a.isContextProvider = oe, a.isElement = ne, a.isForwardRef = ae, a.isFragment = se, a.isLazy = ie, a.isMemo = ce, a.isPortal = fe, a.isProfiler = ue, a.isStrictMode = le, a.isSuspense = me, a.isValidElementType = l, a.typeOf = f;
  })()), a;
}
var q;
function xe() {
  return q || (q = 1, process.env.NODE_ENV === "production" ? M.exports = Pe() : M.exports = Re()), M.exports;
}
var h, U;
function Ce() {
  if (U) return h;
  U = 1;
  var r = xe(), s = {
    childContextTypes: !0,
    contextType: !0,
    contextTypes: !0,
    defaultProps: !0,
    displayName: !0,
    getDefaultProps: !0,
    getDerivedStateFromError: !0,
    getDerivedStateFromProps: !0,
    mixins: !0,
    propTypes: !0,
    type: !0
  }, i = {
    name: !0,
    length: !0,
    prototype: !0,
    caller: !0,
    callee: !0,
    arguments: !0,
    arity: !0
  }, o = {
    $$typeof: !0,
    render: !0,
    defaultProps: !0,
    displayName: !0,
    propTypes: !0
  }, c = {
    $$typeof: !0,
    compare: !0,
    defaultProps: !0,
    displayName: !0,
    propTypes: !0,
    type: !0
  }, y = {};
  y[r.ForwardRef] = o, y[r.Memo] = c;
  function m(d) {
    return r.isMemo(d) ? c : y[d.$$typeof] || s;
  }
  var v = Object.defineProperty, S = Object.getOwnPropertyNames, u = Object.getOwnPropertySymbols, T = Object.getOwnPropertyDescriptor, _ = Object.getPrototypeOf, $ = Object.prototype;
  function E(d, P, x) {
    if (typeof P != "string") {
      if ($) {
        var C = _(P);
        C && C !== $ && E(d, C, x);
      }
      var R = S(P);
      u && (R = R.concat(u(P)));
      for (var l = m(d), f = m(P), e = 0; e < R.length; ++e) {
        var p = R[e];
        if (!i[p] && !(x && x[p]) && !(f && f[p]) && !(l && l[p])) {
          var N = T(P, p);
          try {
            v(d, p, N);
          } catch {
          }
        }
      }
    }
    return d;
  }
  return h = E, h;
}
Ce();
var V = /* @__PURE__ */ b.createContext(
  // we're doing this to avoid preconstruct's dead code elimination in this one case
  // because this module is primarily intended for the browser and node
  // but it's also required in react native and similar environments sometimes
  // and we could have a special build just for that
  // but this is much easier and the native packages
  // might use a different theme context in the future anyway
  typeof HTMLElement < "u" ? /* @__PURE__ */ ve({
    key: "css"
  }) : null
);
V.Provider;
var be = function(s) {
  return /* @__PURE__ */ de(function(i, o) {
    var c = ye(V);
    return s(i, c, o);
  });
}, $e = /* @__PURE__ */ b.createContext({}), O = {}.hasOwnProperty, I = "__EMOTION_TYPE_PLEASE_DO_NOT_USE__", W = function(s, i) {
  var o = {};
  for (var c in i)
    O.call(i, c) && (o[c] = i[c]);
  return o[I] = s, o;
}, Ae = function(s) {
  var i = s.cache, o = s.serialized, c = s.isStringTag;
  return Ee(i, o, c), _e(function() {
    return pe(i, o, c);
  }), null;
}, ge = /* @__PURE__ */ be(function(r, s, i) {
  var o = r.css;
  typeof o == "string" && s.registered[o] !== void 0 && (o = s.registered[o]);
  var c = r[I], y = [o], m = "";
  typeof r.className == "string" ? m = Se(s.registered, y, r.className) : r.className != null && (m = r.className + " ");
  var v = Te(y, void 0, b.useContext($e));
  m += s.key + "-" + v.name;
  var S = {};
  for (var u in r)
    O.call(r, u) && u !== "css" && u !== I && (S[u] = r[u]);
  return S.className = m, i && (S.ref = i), /* @__PURE__ */ b.createElement(b.Fragment, null, /* @__PURE__ */ b.createElement(Ae, {
    cache: s,
    serialized: v,
    isStringTag: typeof c == "string"
  }), /* @__PURE__ */ b.createElement(c, S));
}), H = ge, Ie = A.Fragment, Fe = function(s, i, o) {
  return O.call(i, "css") ? A.jsx(H, W(s, i), o) : A.jsx(s, i, o);
}, Ye = function(s, i, o) {
  return O.call(i, "css") ? A.jsxs(H, W(s, i), o) : A.jsxs(s, i, o);
};
export {
  Ie as F,
  Ye as a,
  Fe as j
};
