// @hakit/core/dist/es/hooks/useFloors/index.js
import "lodash";
import { W as n } from "../../../index-1tDtVGr2.js";
import { useMemo as a } from "react";
import "../useLocale/locales/index.js";
import "home-assistant-js-websocket";
import "@iconify/react";
import "use-debounce";
import { useAreas as l } from "../useAreas/index.js";
import "../../utils/date.js";
function k() {
  const e = l(), r = n((o) => o.floors);
  return a(() => {
    const o = f(e), t = e.filter((s) => !s.floor_id || !o[s.floor_id]), i = Object.values(r).map((s) => ({
      ...s,
      type: "floor",
      areas: o[s.floor_id] || []
    }));
    return t.length && i.push({
      floor_id: "unassigned",
      name: "Unassigned",
      level: null,
      icon: null,
      aliases: [],
      created_at: Date.now(),
      modified_at: Date.now(),
      type: "unassigned",
      areas: t
    }), i;
  }, [e, r]);
}
const f = (e) => {
  const r = {};
  for (const o of e)
    o.floor_id && (o.floor_id in r || (r[o.floor_id] = []), r[o.floor_id].push(o));
  return r;
};
export {
  k as useFloors
};
