// @hakit/core/dist/es/hooks/index.js
import { g as o, u as t, b as s, a, c as i } from "../../index-BiRhtKmJ.js";
import { useEntities as c } from "./useEntities/index.js";
import { createService as m, useService as f } from "./useService/index.js";
import { useLightBrightness as n } from "./useLightBrightness/index.js";
import { useLightTemperature as y } from "./useLightTemperature/index.js";
import { useLightColor as l } from "./useLightColor/index.js";
import { useLowDevices as b } from "./useLowDevices/index.js";
import { useDevice as L } from "./useDevice/index.js";
import { useHistory as S } from "./useHistory/index.js";
import { coordinates as T } from "./useHistory/coordinates.js";
import { useSubscribeEntity as C } from "./useSubscribeEntity/index.js";
import { useLogs as U } from "./useLogs/index.js";
import { useHaStatus as F } from "./useHaStatus/index.js";
import { a1 as O, a0 as N, a8 as h, a6 as k, a5 as B, a7 as P, a9 as H, X as W, a3 as z, a4 as Y, $ as X, a2 as w, _ as G, U as V, W as Z, Y as $, Z as j, V as q } from "../../index-1tDtVGr2.js";
import { useUser as K } from "./useUser/index.js";
import { useUsers as ee } from "./useUsers/index.js";
import { useWeather as oe } from "./useWeather/index.js";
import { getSupportedForecastTypes as se } from "./useWeather/helpers.js";
import { CONTINUOUS_DOMAINS as ie, createHistoricState as ue, localizeStateMessage as ce, localizeTriggerSource as pe, subscribeLogbook as me } from "./useLogs/logbook.js";
import { useAreas as ge } from "./useAreas/index.js";
import { default as xe } from "./useLocale/locales/index.js";
import { areaCompare as Ee, createAreaRegistryEntry as le, deleteAreaRegistryEntry as Ae, getAreaDeviceLookup as be, getAreaEntityLookup as De, subscribeAreaRegistry as Le, updateAreaRegistryEntry as Re } from "../utils/subscribe/areas.js";
import { subscribeEntityRegistry as ve } from "../utils/subscribe/entities.js";
import { subscribeFloorRegistry as _e } from "../utils/subscribe/floors.js";
import { fetchEntitySourcesWithCache as Ie } from "../utils/subscribe/entity_sources.js";
import { DateFormat as de, FirstWeekday as Fe, NumberFormat as Me, findAvailableLanguage as Oe, getUserLocaleLanguage as Ne, subscribeFrontendUserData as he } from "../utils/subscribe/frontend_data.js";
import { subscribeUser as Be, subscribeUsers as Pe, userCollection as He } from "../utils/subscribe/user.js";
import { useTemplate as ze } from "./useTemplate/index.js";
import { useCamera as Xe } from "./useCamera/index.js";
import { useLocaleData as Ge } from "./useLocaleData/index.js";
import { useRegistryData as Ze } from "./useRegistryData/index.js";
import { useFloors as je } from "./useFloors/index.js";
import { ASPECT_RATIO_DEFAULT as Je, CAMERA_ORIENTATIONS as Ke, CAMERA_SUPPORT_ON_OFF as Qe, CAMERA_SUPPORT_STREAM as er, MAX_IMAGE_WIDTH as rr, STREAM_TYPE_HLS as or, STREAM_TYPE_WEB_RTC as tr } from "./useCamera/constants.js";
import { batteryIconByLevel as ar } from "./useIcon/domains/battery.js";
export {
  Je as ASPECT_RATIO_DEFAULT,
  Ke as CAMERA_ORIENTATIONS,
  Qe as CAMERA_SUPPORT_ON_OFF,
  er as CAMERA_SUPPORT_STREAM,
  ie as CONTINUOUS_DOMAINS,
  de as DateFormat,
  Fe as FirstWeekday,
  rr as MAX_IMAGE_WIDTH,
  Me as NumberFormat,
  or as STREAM_TYPE_HLS,
  tr as STREAM_TYPE_WEB_RTC,
  Ee as areaCompare,
  ar as batteryIconByLevel,
  T as coordinates,
  le as createAreaRegistryEntry,
  ue as createHistoricState,
  m as createService,
  Ae as deleteAreaRegistryEntry,
  O as devicesInArea,
  N as fallbackDeviceName,
  Ie as fetchEntitySourcesWithCache,
  Oe as findAvailableLanguage,
  be as getAreaDeviceLookup,
  De as getAreaEntityLookup,
  h as getDeviceContext,
  k as getDeviceEntityDisplayLookup,
  B as getDeviceEntityLookup,
  P as getDeviceIntegrationLookup,
  H as getDevices,
  o as getIconByEntity,
  se as getSupportedForecastTypes,
  Ne as getUserLocaleLanguage,
  xe as locales,
  W as localize,
  ce as localizeStateMessage,
  pe as localizeTriggerSource,
  z as removeConfigEntryFromDevice,
  Y as sortDeviceRegistryByName,
  Le as subscribeAreaRegistry,
  X as subscribeDeviceRegistry,
  ve as subscribeEntityRegistry,
  _e as subscribeFloorRegistry,
  he as subscribeFrontendUserData,
  me as subscribeLogbook,
  Be as subscribeUser,
  Pe as subscribeUsers,
  Re as updateAreaRegistryEntry,
  w as updateDeviceRegistryEntry,
  G as updateLocales,
  ge as useAreas,
  Xe as useCamera,
  V as useConfig,
  L as useDevice,
  c as useEntities,
  t as useEntity,
  je as useFloors,
  F as useHaStatus,
  Z as useHass,
  S as useHistory,
  s as useIcon,
  a as useIconByDomain,
  i as useIconByEntity,
  n as useLightBrightness,
  l as useLightColor,
  y as useLightTemperature,
  $ as useLocale,
  Ge as useLocaleData,
  j as useLocales,
  U as useLogs,
  b as useLowDevices,
  Ze as useRegistryData,
  f as useService,
  q as useStore,
  C as useSubscribeEntity,
  ze as useTemplate,
  K as useUser,
  ee as useUsers,
  oe as useWeather,
  He as userCollection
};
