// @hakit/components/dist/es/Shared/Entity/Climate/ClimateControls/data.js
const r = ["auto", "heat_cool", "heat", "cool", "dry", "fan_only", "off"], f = "°C", _ = "°F", T = "none";
var s = /* @__PURE__ */ ((o) => (o[o.TARGET_TEMPERATURE = 1] = "TARGET_TEMPERATURE", o[o.TARGET_TEMPERATURE_RANGE = 2] = "TARGET_TEMPERATURE_RANGE", o[o.TARGET_HUMIDITY = 4] = "TARGET_HUMIDITY", o[o.FAN_MODE = 8] = "FAN_MODE", o[o.PRESET_MODE = 16] = "PRESET_MODE", o[o.SWING_MODE = 32] = "SWING_MODE", o[o.AUX_HEAT = 64] = "AUX_HEAT", o))(s || {});
const d = r.reduce((o, e, m) => (o[e] = m, o), {}), E = (o, e) => d[o] - d[e], t = {
  cooling: "cool",
  drying: "dry",
  fan: "fan_only",
  preheating: "heat",
  heating: "heat",
  idle: "off",
  off: "off",
  defrosting: "heat"
}, A = {
  cooling: "mdi:snowflake",
  drying: "mdi:water-percent",
  fan: "mdi:fan",
  heating: "mdi:fire",
  idle: "mdi:clock-outline",
  off: "mdi:power",
  preheating: "mdi:heat-wave",
  defrosting: "mdi:snowflake-alert"
}, a = {
  cool: "mdi:snowflake",
  dry: "mdi:water-percent",
  fan_only: "mdi:fan",
  auto: "mdi:thermostat-auto",
  heat: "mdi:fire",
  off: "mdi:power",
  heat_cool: "mdi:sun-snowflake-variant"
}, M = (o) => a[(o ?? "").toLowerCase()], i = {
  away: "mdi:account-arrow-right",
  boost: "mdi:rocket-launch",
  comfort: "mdi:sofa",
  eco: "mdi:leaf",
  home: "mdi:home",
  sleep: "mdi:bed",
  activity: "mdi:motion-sensor"
}, w = (o) => (o ?? "").toLowerCase() in i ? i[o.toLowerCase()] : "mdi:circle-medium", n = {
  on: "mdi:fan",
  off: "mdi:fan-off",
  auto: "mdi:fan-auto",
  low: "mdi:speedometer-slow",
  medium: "mdi:speedometer-medium",
  mid: "mdi:speedometer-medium",
  high: "mdi:speedometer",
  middle: "mdi:speedometer-medium",
  focus: "mdi:target",
  diffuse: "mdi:weather-windy"
}, I = (o) => (o ?? "").toLowerCase() in n ? n[o.toLowerCase()] : "mdi:circle-medium", c = {
  on: "mdi:arrow-oscillating",
  off: "mdi:arrow-oscillating-off",
  vertical: "mdi:arrow-up-down",
  horizontal: "mdi:arrow-left-right",
  both: "mdi:arrow-all"
}, l = (o) => (o ?? "").toLowerCase() in c ? c[o.toLowerCase()] : "mdi:circle-medium";
export {
  n as CLIMATE_FAN_MODE_ICONS,
  A as CLIMATE_HVAC_ACTION_ICONS,
  t as CLIMATE_HVAC_ACTION_TO_MODE,
  a as CLIMATE_HVAC_MODE_ICONS,
  i as CLIMATE_PRESET_MODE_ICONS,
  T as CLIMATE_PRESET_NONE,
  c as CLIMATE_SWING_MODE_ICONS,
  s as ClimateEntityFeature,
  f as UNIT_C,
  _ as UNIT_F,
  E as compareClimateHvacModes,
  I as computeFanModeIcon,
  M as computeHvacModeIcon,
  w as computePresetModeIcon,
  l as computeSwingModeIcon
};
