// @hakit/core/dist/es/utils/colors/convert-color.js
import { expandHex as _ } from "./hex.js";
const b = (t) => {
  const s = Math.round(Math.min(Math.max(t, 0), 255)).toString(16);
  return s.length === 1 ? `0${s}` : s;
}, S = (t) => (t = _(t), [parseInt(t.substring(0, 2), 16), parseInt(t.substring(2, 4), 16), parseInt(t.substring(4, 6), 16)]), f = (t) => `#${b(t[0])}${b(t[1])}${b(t[2])}`, h = 0.95047, i = 1, l = 1.08883, m = 0.137931034, v = 0.206896552, y = 0.12841855, N = 8856452e-9, a = (t) => (t /= 255, t <= 0.04045 ? t / 12.92 : ((t + 0.055) / 1.055) ** 2.4), g = (t) => t > N ? t ** (1 / 3) : t / y + m, u = (t) => 255 * (t <= 304e-5 ? 12.92 * t : 1.055 * t ** (1 / 2.4) - 0.055), x = (t) => t > v ? t * t * t : y * (t - m), $ = (t) => {
  let [s, r, o] = t;
  s = a(s), r = a(r), o = a(o);
  const n = g((0.4124564 * s + 0.3575761 * r + 0.1804375 * o) / h), c = g((0.2126729 * s + 0.7151522 * r + 0.072175 * o) / i), e = g((0.0193339 * s + 0.119192 * r + 0.9503041 * o) / l);
  return [n, c, e];
}, X = (t) => {
  const [s, r, o] = $(t), n = 116 * r - 16;
  return [n < 0 ? 0 : n, 500 * (s - r), 200 * (r - o)];
}, I = (t) => {
  const [s, r, o] = t;
  let n = (s + 16) / 116, c = isNaN(r) ? n : n + r / 500, e = isNaN(o) ? n : n - o / 200;
  n = i * x(n), c = h * x(c), e = l * x(e);
  const z = u(3.2404542 * c - 1.5371385 * n - 0.4985314 * e), M = u(-0.969266 * c + 1.8760108 * n + 0.041556 * e), p = u(0.0556434 * c - 0.2040259 * n + 1.0572252 * e);
  return [z, M, p];
}, Y = (t) => {
  const s = I(t);
  return f(s);
}, d = (t) => {
  const [s, r, o] = t, n = Math.max(s, r, o), c = n - Math.min(s, r, o), e = c && (n === s ? (r - o) / c : n === r ? 2 + (o - s) / c : 4 + (s - r) / c);
  return [60 * (e < 0 ? e + 6 : e), n && c / n, n];
}, k = (t) => {
  const [s, r, o] = t, n = (c) => {
    const e = (c + s / 60) % 6;
    return o - o * r * Math.max(Math.min(e, 4 - e, 1), 0);
  };
  return [n(5), n(3), n(1)];
}, Z = (t) => d(t).slice(0, 2), j = (t) => k([t[0], t[1], 255]);
export {
  S as hex2rgb,
  j as hs2rgb,
  k as hsv2rgb,
  Y as lab2hex,
  I as lab2rgb,
  f as rgb2hex,
  Z as rgb2hs,
  d as rgb2hsv,
  X as rgb2lab
};
