// @hakit/core/dist/es/hooks/useAreas/index.js
import "lodash";
import { W as s } from "../../../index-1tDtVGr2.js";
import { useMemo as y } from "react";
import "../useLocale/locales/index.js";
import "home-assistant-js-websocket";
import "@iconify/react";
import "use-debounce";
import "../../utils/date.js";
function S() {
  const { joinHassUrl: n } = s.getState().helpers, r = s((e) => e.areas), o = s((e) => e.devices), c = s((e) => e.floors), a = s((e) => e.entitiesRegistryDisplay), d = s((e) => e.entities);
  return y(() => {
    const e = Object.values(r);
    return v({
      areas: e,
      devices: o,
      entitiesRegistryDisplay: a,
      floors: c,
      joinHassUrl: n,
      hassEntities: d
    });
  }, [r, o, n, a, c, d]);
}
function v({ areas: n, devices: r, entitiesRegistryDisplay: o, floors: c, joinHassUrl: a, hassEntities: d }) {
  const e = (i) => {
    const p = [];
    for (const t of Object.values(r))
      t.area_id === i.area_id && t.entry_type !== "service" && p.push(t);
    const u = /* @__PURE__ */ new Set(), l = [];
    for (const t of Object.values(o)) {
      const _ = t.area_id === i.area_id, m = !t.area_id && t.device_id && r[t.device_id]?.area_id === i.area_id;
      if (_ || m) {
        const f = d[t.entity_id];
        f && !u.has(t.entity_id) && (u.add(t.entity_id), l.push(f));
      }
    }
    return {
      ...i,
      picture: i.picture ? a(i.picture) : null,
      devices: p,
      entities: l,
      floor: i.floor_id && c[i.floor_id] || null
    };
  };
  return n.map(e);
}
export {
  S as useAreas
};
