// @hakit/core/dist/es/index.js
import { rgbw2rgb as r, rgbww2rgb as o, temperature2rgb as a } from "./utils/colors/convert-light-color.js";
import { hex2rgb as i, hs2rgb as m, hsv2rgb as p, lab2hex as n, lab2rgb as u, rgb2hex as f, rgb2hs as y, rgb2hsv as g, rgb2lab as c } from "./utils/colors/convert-color.js";
import { COLORS as l, getColorByIndex as T } from "./utils/colors/colors.js";
import { THEME_COLORS as D, computeCssColor as A } from "./utils/colors/compute-color.js";
import { expandHex as b, hexBlend as N } from "./utils/colors/hex.js";
import { getRGBContrastRatio as C, luminosity as h, rgbContrast as I } from "./utils/colors/rgb.js";
import { stateColorBrightness as _ } from "./utils/colors/index.js";
import { labBrighten as O, labDarken as v } from "./utils/colors/lab.js";
import { D as U, N as M, O as B, H as k, T as W, r as P, Q as H, e as Y, n as V, m as w, p as z, o as G, v as Z, t as j, y as K, q as X, f as q, R as J, S as Q, i as $, h as ee, a1 as te, I as re, J as oe, u as ae, a0 as se, E as ie, x as me, w as pe, L as ne, a8 as ue, a6 as fe, a5 as ye, a7 as ge, a9 as ce, M as Ee, j as le, k as Te, K as xe, A as De, z as Ae, g as Se, P as be, a as Ne, c as Re, b as Ce, l as he, d as Ie, X as de, a3 as _e, C as Le, a4 as Oe, G as ve, s as Fe, $ as Ue, F as Me, a2 as Be, B as ke, _ as We, U as Pe, W as He, Y as Ye, Z as Ve, V as we } from "../index-1tDtVGr2.js";
import { getCssColorValue as Ge } from "./utils/colors.js";
import { computeDomain as je } from "./utils/computeDomain.js";
import { computeDomainTitle as Xe } from "./utils/computeDomainTitle.js";
import { supportsFeatureFromAttributes as Je } from "./utils/supports-feature.js";
import { timeAgo as $e } from "./utils/time/time-ago.js";
import { caseInsensitiveStringCompare as tt, stringCompare as rt, toReadableString as ot } from "./utils/string.js";
import { formatNumber as st, getDefaultFormatOptions as it, getNumberFormatOptions as mt, isNumericFromAttributes as pt, isNumericState as nt, round as ut } from "./utils/number.js";
import { LOCAL_TIME_ZONE as yt, TimeFormat as gt, TimeZone as ct, UNIT_TO_MILLISECOND_CONVERT as Et, checkValidDate as lt, formatAmPmSuffix as Tt, formatDate as xt, formatDateMonth as Dt, formatDateMonthYear as At, formatDateNumeric as St, formatDateShort as bt, formatDateTime as Nt, formatDateTimeNumeric as Rt, formatDateTimeWithBrowserDefaults as Ct, formatDateTimeWithSeconds as ht, formatDateVeryShort as It, formatDateWeekday as dt, formatDateWeekdayDay as _t, formatDateWeekdayShort as Lt, formatDateYear as Ot, formatDuration as vt, formatHour as Ft, formatMinute as Ut, formatSeconds as Mt, formatShortDateTime as Bt, formatShortDateTimeWithConditionalYear as kt, formatShortDateTimeWithYear as Wt, formatTime as Pt, formatTimeWithoutAmPm as Ht, isDate as Yt, isTimestamp as Vt, millisecondsToDuration as wt, resolveTimeZone as zt, shouldUseAmPm as Gt } from "./utils/date.js";
import { stripPrefixFromEntityName as jt } from "./utils/stripPrefixFromEntityName.js";
import { DEFAULT_STATES as Xt, DOMAINS_WITH_DYNAMIC_PICTURE as qt, OFF as Jt, OFF_STATES as Qt, ON as $t, UNAVAILABLE as er, UNAVAILABLE_STATES as tr, UNIT_C as rr, UNIT_F as or, UNKNOWN as ar, isOffState as sr, isUnavailableState as ir } from "./data/entity.js";
import { loadTokens as pr, saveTokens as nr } from "./HassConnect/token-storage.js";
import { g as fr, u as yr, b as gr, a as cr, c as Er } from "../index-BiRhtKmJ.js";
import { useEntities as Tr } from "./hooks/useEntities/index.js";
import { createService as Dr, useService as Ar } from "./hooks/useService/index.js";
import { useLightBrightness as br } from "./hooks/useLightBrightness/index.js";
import { useLightTemperature as Rr } from "./hooks/useLightTemperature/index.js";
import { useLightColor as hr } from "./hooks/useLightColor/index.js";
import { useLowDevices as dr } from "./hooks/useLowDevices/index.js";
import { useDevice as Lr } from "./hooks/useDevice/index.js";
import { useHistory as vr } from "./hooks/useHistory/index.js";
import { coordinates as Ur } from "./hooks/useHistory/coordinates.js";
import { useSubscribeEntity as Br } from "./hooks/useSubscribeEntity/index.js";
import { useLogs as Wr } from "./hooks/useLogs/index.js";
import { useHaStatus as Hr } from "./hooks/useHaStatus/index.js";
import { useUser as Vr } from "./hooks/useUser/index.js";
import { useUsers as zr } from "./hooks/useUsers/index.js";
import { useWeather as Zr } from "./hooks/useWeather/index.js";
import { getSupportedForecastTypes as Kr } from "./hooks/useWeather/helpers.js";
import { CONTINUOUS_DOMAINS as qr, createHistoricState as Jr, localizeStateMessage as Qr, localizeTriggerSource as $r, subscribeLogbook as eo } from "./hooks/useLogs/logbook.js";
import { useAreas as ro } from "./hooks/useAreas/index.js";
import { default as ao } from "./hooks/useLocale/locales/index.js";
import { areaCompare as io, createAreaRegistryEntry as mo, deleteAreaRegistryEntry as po, getAreaDeviceLookup as no, getAreaEntityLookup as uo, subscribeAreaRegistry as fo, updateAreaRegistryEntry as yo } from "./utils/subscribe/areas.js";
import { subscribeEntityRegistry as co } from "./utils/subscribe/entities.js";
import { subscribeFloorRegistry as lo } from "./utils/subscribe/floors.js";
import { fetchEntitySourcesWithCache as xo } from "./utils/subscribe/entity_sources.js";
import { DateFormat as Ao, FirstWeekday as So, NumberFormat as bo, findAvailableLanguage as No, getUserLocaleLanguage as Ro, subscribeFrontendUserData as Co } from "./utils/subscribe/frontend_data.js";
import { subscribeUser as Io, subscribeUsers as _o, userCollection as Lo } from "./utils/subscribe/user.js";
import { useTemplate as vo } from "./hooks/useTemplate/index.js";
import { useCamera as Uo } from "./hooks/useCamera/index.js";
import { useLocaleData as Bo } from "./hooks/useLocaleData/index.js";
import { useRegistryData as Wo } from "./hooks/useRegistryData/index.js";
import { useFloors as Ho } from "./hooks/useFloors/index.js";
import { ASPECT_RATIO_DEFAULT as Vo, CAMERA_ORIENTATIONS as wo, CAMERA_SUPPORT_ON_OFF as zo, CAMERA_SUPPORT_STREAM as Go, MAX_IMAGE_WIDTH as Zo, STREAM_TYPE_HLS as jo, STREAM_TYPE_WEB_RTC as Ko } from "./hooks/useCamera/constants.js";
import { LIGHT_COLOR_MODES as qo } from "./types/autogenerated-types-by-domain.js";
import { batteryIconByLevel as Qo } from "./hooks/useIcon/domains/battery.js";
export {
  Vo as ASPECT_RATIO_DEFAULT,
  wo as CAMERA_ORIENTATIONS,
  zo as CAMERA_SUPPORT_ON_OFF,
  Go as CAMERA_SUPPORT_STREAM,
  l as COLORS,
  qr as CONTINUOUS_DOMAINS,
  U as DEFAULT_ENTITY_NAME,
  Xt as DEFAULT_STATES,
  qt as DOMAINS_WITH_DYNAMIC_PICTURE,
  M as DOMAIN_ATTRIBUTES_FORMATTERS,
  B as DOMAIN_ATTRIBUTES_UNITS,
  Ao as DateFormat,
  So as FirstWeekday,
  k as HassConnect,
  qo as LIGHT_COLOR_MODES,
  yt as LOCAL_TIME_ZONE,
  Zo as MAX_IMAGE_WIDTH,
  bo as NumberFormat,
  Jt as OFF,
  Qt as OFF_STATES,
  $t as ON,
  jo as STREAM_TYPE_HLS,
  Ko as STREAM_TYPE_WEB_RTC,
  W as TEMPERATURE_ATTRIBUTES,
  D as THEME_COLORS,
  gt as TimeFormat,
  ct as TimeZone,
  er as UNAVAILABLE,
  tr as UNAVAILABLE_STATES,
  rr as UNIT_C,
  or as UNIT_F,
  Et as UNIT_TO_MILLISECOND_CONVERT,
  ar as UNKNOWN,
  io as areaCompare,
  Qo as batteryIconByLevel,
  tt as caseInsensitiveStringCompare,
  lt as checkValidDate,
  P as computeAreaName,
  H as computeAttributeValueDisplay,
  A as computeCssColor,
  Y as computeDefaultFavoriteColors,
  V as computeDeviceName,
  w as computeDeviceNameDisplay,
  je as computeDomain,
  Xe as computeDomainTitle,
  z as computeEntityEntryName,
  G as computeEntityName,
  Z as computeEntityNameDisplay,
  j as computeEntityNameList,
  K as computeEntityRegistryName,
  X as computeFloorName,
  q as computeObjectId,
  J as computeStateDisplay,
  Q as computeStateDisplayFromEntityAttributes,
  $ as computeStateName,
  ee as computeStateNameFromEntityAttributes,
  Ur as coordinates,
  mo as createAreaRegistryEntry,
  Jr as createHistoricState,
  Dr as createService,
  po as deleteAreaRegistryEntry,
  te as devicesInArea,
  re as entityRegistryByEntityId,
  oe as entityRegistryById,
  ae as entityUseDeviceName,
  b as expandHex,
  se as fallbackDeviceName,
  ie as fetchEntityRegistryDisplay,
  xo as fetchEntitySourcesWithCache,
  No as findAvailableLanguage,
  me as findBatteryChargingEntity,
  pe as findBatteryEntity,
  Tt as formatAmPmSuffix,
  xt as formatDate,
  Dt as formatDateMonth,
  At as formatDateMonthYear,
  St as formatDateNumeric,
  bt as formatDateShort,
  Nt as formatDateTime,
  Rt as formatDateTimeNumeric,
  Ct as formatDateTimeWithBrowserDefaults,
  ht as formatDateTimeWithSeconds,
  It as formatDateVeryShort,
  dt as formatDateWeekday,
  _t as formatDateWeekdayDay,
  Lt as formatDateWeekdayShort,
  Ot as formatDateYear,
  vt as formatDuration,
  Ft as formatHour,
  Ut as formatMinute,
  st as formatNumber,
  Mt as formatSeconds,
  Bt as formatShortDateTime,
  kt as formatShortDateTimeWithConditionalYear,
  Wt as formatShortDateTimeWithYear,
  Pt as formatTime,
  Ht as formatTimeWithoutAmPm,
  no as getAreaDeviceLookup,
  uo as getAreaEntityLookup,
  ne as getAutomaticEntityIds,
  T as getColorByIndex,
  Ge as getCssColorValue,
  it as getDefaultFormatOptions,
  ue as getDeviceContext,
  fe as getDeviceEntityDisplayLookup,
  ye as getDeviceEntityLookup,
  ge as getDeviceIntegrationLookup,
  ce as getDevices,
  Ee as getEntities,
  le as getEntityContext,
  Te as getEntityEntryContext,
  xe as getEntityPlatformLookup,
  De as getExtendedEntityRegistryEntries,
  Ae as getExtendedEntityRegistryEntry,
  fr as getIconByEntity,
  Se as getLightCurrentModeRgbColor,
  mt as getNumberFormatOptions,
  C as getRGBContrastRatio,
  Kr as getSupportedForecastTypes,
  Ro as getUserLocaleLanguage,
  be as getWeatherUnit,
  i as hex2rgb,
  N as hexBlend,
  m as hs2rgb,
  p as hsv2rgb,
  Yt as isDate,
  pt as isNumericFromAttributes,
  nt as isNumericState,
  sr as isOffState,
  Vt as isTimestamp,
  ir as isUnavailableState,
  n as lab2hex,
  u as lab2rgb,
  O as labBrighten,
  v as labDarken,
  Ne as lightIsInColorMode,
  Re as lightSupportsBrightness,
  Ce as lightSupportsColor,
  he as lightSupportsColorMode,
  Ie as lightSupportsFavoriteColors,
  pr as loadTokens,
  ao as locales,
  de as localize,
  Qr as localizeStateMessage,
  $r as localizeTriggerSource,
  h as luminosity,
  wt as millisecondsToDuration,
  _e as removeConfigEntryFromDevice,
  Le as removeEntityRegistryEntry,
  zt as resolveTimeZone,
  f as rgb2hex,
  y as rgb2hs,
  g as rgb2hsv,
  c as rgb2lab,
  I as rgbContrast,
  r as rgbw2rgb,
  o as rgbww2rgb,
  ut as round,
  nr as saveTokens,
  Gt as shouldUseAmPm,
  Oe as sortDeviceRegistryByName,
  ve as sortEntityRegistryByName,
  Fe as stateActive,
  _ as stateColorBrightness,
  rt as stringCompare,
  jt as stripPrefixFromEntityName,
  fo as subscribeAreaRegistry,
  Ue as subscribeDeviceRegistry,
  co as subscribeEntityRegistry,
  Me as subscribeEntityRegistryDisplay,
  lo as subscribeFloorRegistry,
  Co as subscribeFrontendUserData,
  eo as subscribeLogbook,
  Io as subscribeUser,
  _o as subscribeUsers,
  Je as supportsFeatureFromAttributes,
  a as temperature2rgb,
  $e as timeAgo,
  ot as toReadableString,
  yo as updateAreaRegistryEntry,
  Be as updateDeviceRegistryEntry,
  ke as updateEntityRegistryEntry,
  We as updateLocales,
  ro as useAreas,
  Uo as useCamera,
  Pe as useConfig,
  Lr as useDevice,
  Tr as useEntities,
  yr as useEntity,
  Ho as useFloors,
  Hr as useHaStatus,
  He as useHass,
  vr as useHistory,
  gr as useIcon,
  cr as useIconByDomain,
  Er as useIconByEntity,
  br as useLightBrightness,
  hr as useLightColor,
  Rr as useLightTemperature,
  Ye as useLocale,
  Bo as useLocaleData,
  Ve as useLocales,
  Wr as useLogs,
  dr as useLowDevices,
  Wo as useRegistryData,
  Ar as useService,
  we as useStore,
  Br as useSubscribeEntity,
  vo as useTemplate,
  Vr as useUser,
  zr as useUsers,
  Zr as useWeather,
  Lo as userCollection
};
