// @hakit/core/dist/es/hooks/useLightColor/index.js
import { useMemo as s } from "react";
import "lodash";
import { rgb2hs as v } from "../../utils/colors/convert-color.js";
import { l, g } from "../../../index-1tDtVGr2.js";
import "../useLocale/locales/index.js";
import "home-assistant-js-websocket";
import "@iconify/react";
import "use-debounce";
import "../../utils/date.js";
import { ON as u } from "../../data/entity.js";
import { LIGHT_COLOR_MODES as a } from "../../types/autogenerated-types-by-domain.js";
const C = (o) => {
  const c = s(() => {
    if (o.state === u) {
      if (o.attributes.color_mode === a.RGB && o.attributes.rgb_color && !l(o, a.RGBWW) && !l(o, a.RGBW)) {
        const r = Math.max(...o.attributes.rgb_color);
        if (r < 255)
          return r;
      }
      return;
    }
  }, [o]), d = s(() => {
    const r = o.state !== u ? void 0 : o.attributes.color_mode === a.RGBW && o.attributes.rgbw_color ? Math.round(o.attributes.rgbw_color[3] * 100 / 255) : void 0;
    return r != null ? r * 255 / 100 : void 0;
  }, [o]), e = s(() => {
    const r = o.state !== u ? void 0 : o.attributes.color_mode === a.RGBWW && o.attributes.rgbww_color ? Math.round(o.attributes.rgbww_color[3] * 100 / 255) : void 0;
    return r != null ? r * 255 / 100 : void 0;
  }, [o]), i = s(() => {
    const r = o.state !== u ? void 0 : o.attributes.color_mode === a.RGBWW && o.attributes.rgbww_color ? Math.round(o.attributes.rgbww_color[4] * 100 / 255) : void 0;
    return r != null ? r * 255 / 100 : void 0;
  }, [o]), t = g(o), m = s(() => {
    const r = o.state !== u ? void 0 : t ? Math.round(Math.max(...t.slice(0, 3)) * 100 / 255) : void 0;
    return r != null ? r * 255 / 100 : void 0;
  }, [o, t]), b = s(() => o.state !== u ? void 0 : t ? v(t.slice(0, 3)) : void 0, [o, t]);
  return {
    brightnessAdjusted: c,
    white: d,
    coolWhite: e,
    warmWhite: i,
    colorBrightness: m,
    hs: b
  };
};
export {
  C as useLightColor
};
