// @hakit/components/dist/es/Cards/EntitiesCard/action-types/climate.js
import { a as r, j as l, F as s } from "../../../../emotion-react-jsx-runtime.browser.esm-BErL7L7Y.js";
import { c as f } from "../../../../emotion-styled-base.browser.esm-Dyg8Y4aq.js";
import { isUnavailableState as _, useHass as h, localize as o, OFF as b } from "@hakit/core";
import { CLIMATE_PRESET_NONE as g } from "../../../Shared/Entity/Climate/ClimateControls/data.js";
function v() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}
const w = /* @__PURE__ */ f("div", process.env.NODE_ENV === "production" ? {
  target: "e1ri2dm00"
} : {
  target: "e1ri2dm00",
  label: "Wrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "fz81w7",
  styles: "display:flex;flex-direction:column;justify-content:center;white-space:nowrap;.target,.current{display:flex;flex-direction:row;flex-wrap:nowrap;gap:0.25rem;justify-content:flex-end;}.target{color:var(--ha-S50-contrast);}.current{color:var(--ha-S500-contrast);}.state-label{font-weight:bold;}.unit{}"
} : {
  name: "fz81w7",
  styles: "display:flex;flex-direction:column;justify-content:center;white-space:nowrap;.target,.current{display:flex;flex-direction:row;flex-wrap:nowrap;gap:0.25rem;justify-content:flex-end;}.target{color:var(--ha-S50-contrast);}.current{color:var(--ha-S500-contrast);}.state-label{font-weight:bold;}.unit{}",
  toString: v
});
function $({
  entity: t,
  className: c,
  ...m
}) {
  const u = _(t.state), e = h((a) => a.formatter), d = () => {
    if (t.attributes.current_temperature != null && t.attributes.current_humidity != null)
      return `${e.attributeValue(t, "current_temperature")}/
      ${e.attributeValue(t, "current_humidity")}`;
    if (t.attributes.current_temperature != null)
      return e.attributeValue(t, "current_temperature");
    if (t.attributes.current_humidity != null)
      return e.attributeValue(t, "current_humidity");
  }, i = () => {
    if (u)
      return o("unavailable");
    const a = e.stateValue(t);
    return t.attributes.hvac_action && t.state !== b ? `${e.attributeValue(t, "hvac_action")} (${a})` : a;
  }, p = () => t.attributes.target_temp_low != null && t.attributes.target_temp_high != null ? `${e.attributeValue(t, "target_temp_low")}-${e.attributeValue(t, "target_temp_high")}` : t.attributes.temperature != null ? e.attributeValue(t, "temperature") : t.attributes.target_humidity_low != null && t.attributes.target_humidity_high != null ? `${e.attributeValue(t, "target_humidity_low")}-${e.attributeValue(t, "target_humidity_high")}` : t.attributes.humidity != null ? e.attributeValue(t, "humidity") : "", n = d();
  return /* @__PURE__ */ r(w, { className: `action-type-climate ${c}`, ...m, children: [
    /* @__PURE__ */ l("div", { className: "target", children: u ? i() : /* @__PURE__ */ r(s, { children: [
      /* @__PURE__ */ r("span", { className: "state-label", children: [
        i(),
        t.attributes.preset_mode && t.attributes.preset_mode !== g ? /* @__PURE__ */ r(s, { children: [
          "- ",
          e.attributeValue(t, "preset_mode")
        ] }) : null
      ] }),
      /* @__PURE__ */ l("div", { className: "unit", children: p() })
    ] }) }),
    n && !u ? /* @__PURE__ */ r("div", { className: "current", children: [
      o("currently"),
      ":",
      /* @__PURE__ */ l("div", { className: "unit", children: n })
    ] }) : null
  ] });
}
export {
  $ as default
};
