// @hakit/components/dist/es/Cards/EntitiesCard/action-types/sensor.js
import { j as e } from "../../../../emotion-react-jsx-runtime.browser.esm-BErL7L7Y.js";
import { c as n } from "../../../../emotion-styled-base.browser.esm-Dyg8Y4aq.js";
import { useHass as a } from "@hakit/core";
function i() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}
const s = /* @__PURE__ */ n("div", process.env.NODE_ENV === "production" ? {
  target: "e97anlh0"
} : {
  target: "e97anlh0",
  label: "Wrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "jy7joq",
  styles: "display:flex;flex-direction:column;justify-content:center;white-space:nowrap;div{text-align:right;}.value{direction:ltr;}"
} : {
  name: "jy7joq",
  styles: "display:flex;flex-direction:column;justify-content:center;white-space:nowrap;div{text-align:right;}.value{direction:ltr;}",
  toString: i
});
function p({
  entity: t
}) {
  const r = a((o) => o.formatter);
  return /* @__PURE__ */ e(s, { children: /* @__PURE__ */ e("div", { className: "text-content value", children: r.stateValue(t) }) });
}
export {
  p as default
};
