// @hakit/components/dist/es/Cards/EntitiesCard/action-types/toggle.js
import { j as i } from "../../../../emotion-react-jsx-runtime.browser.esm-BErL7L7Y.js";
import { c as f } from "../../../../emotion-styled-base.browser.esm-Dyg8Y4aq.js";
import { useHass as S, isUnavailableState as b, ON as n, OFF as N, computeDomain as w } from "@hakit/core";
import { useRef as k, useState as E, useEffect as y } from "react";
import I from "react-switch";
function d() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}
const T = /* @__PURE__ */ f("div", process.env.NODE_ENV === "production" ? {
  target: "e1hc5dsg1"
} : {
  target: "e1hc5dsg1",
  label: "Wrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "1e2z4ty",
  styles: "display:flex;flex-direction:column;justify-content:center;white-space:nowrap"
} : {
  name: "1e2z4ty",
  styles: "display:flex;flex-direction:column;justify-content:center;white-space:nowrap",
  toString: d
}), x = /* @__PURE__ */ f(I, process.env.NODE_ENV === "production" ? {
  target: "e1hc5dsg0"
} : {
  target: "e1hc5dsg0",
  label: "StyledSwitch"
})(process.env.NODE_ENV === "production" ? {
  name: "88invf",
  styles: "color:var(--ha-100);.react-switch-bg{background:var(--ha-S100)!important;}&.checked{.react-switch-bg{background:var(--ha-A400)!important;}}"
} : {
  name: "88invf",
  styles: "color:var(--ha-100);.react-switch-bg{background:var(--ha-S100)!important;}&.checked{.react-switch-bg{background:var(--ha-A400)!important;}}",
  toString: d
});
function C({
  entity: e,
  className: p,
  ...h
}) {
  const t = k(null), m = S((r) => r.formatter), l = b(e.state), v = e.state === n || e.state === N || l, [o, a] = E(e.state === n), g = async (r) => {
    const s = w(e.entity_id);
    let c;
    s === "lock" ? c = r ? "unlock" : "lock" : s === "cover" ? c = r ? "open_cover" : "close_cover" : s === "valve" ? c = r ? "open_valve" : "close_valve" : c = r ? "turn_on" : "turn_off";
    const u = o;
    a(r), t.current && clearTimeout(t.current), t.current = setTimeout(() => {
      a(u);
    }, 2e3);
    try {
      await e.service[c](), t.current && clearTimeout(t.current), t.current = null;
    } catch (_) {
      console.error("Error calling service", _), t.current && clearTimeout(t.current), t.current = null, a(u);
    }
  };
  return y(() => {
    t.current || o === (e.state === n) || a(e.state === n);
  }, [e.state, o]), /* @__PURE__ */ i(T, { className: `action-type-toggle ${p}`, ...h, children: v ? /* @__PURE__ */ i(
    x,
    {
      disabled: l,
      checked: o,
      className: `${o ? "checked" : ""}`,
      onColor: "#fff",
      offColor: "#fff",
      checkedIcon: !1,
      uncheckedIcon: !1,
      height: 20,
      width: 37,
      onChange: () => g(!o)
    }
  ) : /* @__PURE__ */ i("div", { className: "text-content", children: m.stateValue(e) }) });
}
export {
  C as default
};
