// @hakit/components/dist/es/Shared/Column/index.js
import { j as e } from "../../../emotion-react-jsx-runtime.browser.esm-BErL7L7Y.js";
import { c as l } from "../../../emotion-styled-base.browser.esm-Dyg8Y4aq.js";
import { css as i } from "@emotion/react";
import a from "@emotion/is-prop-valid";
const n = /* @__PURE__ */ l("div", {
  shouldForwardProp: (t) => a(t),
  target: "e1vx4js00"
})("display:flex;flex-direction:column;flex-wrap:", ({
  wrap: t
}) => t || "wrap", ";justify-content:", ({
  justifyContent: t
}) => t || "center", ";align-items:", ({
  alignItems: t
}) => t || "center", ";", ({
  justifyContent: t
}) => `
    --stretch: ${t === "stretch" ? "100%" : "false"};
  `, " ", (t) => typeof t.gap == "string" && `
    gap: ${t.gap};
    --gap: ${t.gap ?? "0px"};
  `, " ", (t) => t.fullHeight && "height: 100%;", " ", (t) => t.fullWidth && "width: 100%;", ";");
function m(t) {
  return /* @__PURE__ */ e(n, { ...t, css: /* @__PURE__ */ i(t.cssStyles ?? "", ";" + (process.env.NODE_ENV === "production" ? "" : ";label:Column;")), className: `${t.className ?? ""} ${t.fullHeight ? "full-height" : ""} ${t.fullWidth ? "full-width" : ""} ${t.justifyContent ? t.justifyContent : "center"} ${t.alignItems ? t.alignItems : "center"} ${t.wrap ? t.wrap : "wrap"}` });
}
export {
  m as Column
};
