// @hakit/components/dist/es/Cards/TimeCard/shared.js
import { c as t } from "../../../emotion-styled-base.browser.esm-Dyg8Y4aq.js";
function e() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}
const a = /* @__PURE__ */ t("span", process.env.NODE_ENV === "production" ? {
  target: "e1mj7xtk1"
} : {
  target: "e1mj7xtk1",
  label: "Time"
})(process.env.NODE_ENV === "production" ? {
  name: "13nmimj",
  styles: "all:unset;font-family:var(--ha-font-family);font-size:2rem;color:var(--ha-S200-contrast);font-weight:400;display:flex;flex-direction:row"
} : {
  name: "13nmimj",
  styles: "all:unset;font-family:var(--ha-font-family);font-size:2rem;color:var(--ha-S200-contrast);font-weight:400;display:flex;flex-direction:row",
  toString: e
}), n = /* @__PURE__ */ t("span", process.env.NODE_ENV === "production" ? {
  target: "e1mj7xtk0"
} : {
  target: "e1mj7xtk0",
  label: "AmOrPm"
})(process.env.NODE_ENV === "production" ? {
  name: "tgy6w2",
  styles: "all:unset;font-family:var(--ha-font-family);font-size:2rem;color:var(--ha-S400-contrast);font-weight:300"
} : {
  name: "tgy6w2",
  styles: "all:unset;font-family:var(--ha-font-family);font-size:2rem;color:var(--ha-S400-contrast);font-weight:300",
  toString: e
});
export {
  n as AmOrPm,
  a as Time
};
